torchmultimodal/modules/layers/text_embedding.py

```diff
@@ -0,0 +1,104 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import Optional
+
+import torch
+from torch import nn, Tensor
+
+
+class TextEmbeddings(nn.Module):
+    """Construct word, position, and token type embeddings following BERT.
+
+    Attributes:
+        hidden_size (int): size of embedding space. Default is 768.
+        vocab_size (int): size of vocabulary. Default is 30522.
```

Comment:
If this is supposed to be a horizontal components, do we still need to set default values (I assume they all come from a specific model right?) So to be clear to the users, we could either:

- Defer those default values to the model that uses this component and keeping it non-opinionated here.
- Rename this as `BERTTextEmbeddings` to avoid over generalization